server/src/routers/converterRouter.js

```diff
@@ -1,46 +1,58 @@
 const express = require('express');
 const multer = require('multer');
-const upload = multer();
 const router = new express.Router();
 
 const { mapping } = require('../constants/config');
-const { parseInputFile, sendResponse } = require('../util');
+const { parseInput, sendResponse } = require('../util');
 
 router.use(express.json());
 
 var Converter = require('api-spec-converter');
+var upload = multer().single('file');
 
-router.post('/api/specification/:format/:convertTo', upload.single('file'), (req, res) => {
-    const {format, convertTo} = req.params,
-        {targetSyntax = 'json', toFile = false} = req.query;
+router.post('/api/specification/:format/:convertTo', (req, res) => {
+  const {format, convertTo} = req.params,
+      {targetSyntax = 'json', toFile = false} = req.query;
 
-    if(!(mapping[format] && mapping[format].includes(convertTo))) {
-      return res.status(404).send({'message': 'Conversion mapping not found.'});
+  upload(req, res, function() {
+    if (req.fileValidationError) {
+      return res.status(500).send({'message': req.fileValidationError});
     }
-    const file = req.file;
-
-    if(!file) {
-      return res.status(400).send({'message': 'File not available.'});
+    else if (!req.file && ((typeof req.body === 'string' && req.body === '') || (typeof req.body === 'object' && Object.keys(req.body).length === 0))) {
```

Comment:
Optionally, we can use [lodash's isEmpty](https://lodash.com/docs/4.17.15#isEmpty) function here